### DISCUSSION TITLE
Ender 3 Max Neo G-code configuration file

### DISCUSSION THREAD

**Discussion by KeithCu** [mriscoc/Ender3V2S1#519] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have an Ender 3 v2 and it's working great with this firmware, but I'm thinking of buying an Ender 3 Max Neo, which has the important upgrades an Ev2 would want, plus a larger bed size, and it's on sale:
https://store.creality.com/products/ender-3-max-neo-3d-printer?spm=..page_2510923.header_1.1

I see it is running a 4.2.2 board and doesn't use a touch screen, so it should be fine. However, I see there are quite some settings to change to get a larger bed size. It would be nice if it had an easy support for the 300mm printers with a custom build or something.

It would also be nice if the website listed on the front page the set of Creality printers that should work out of the box. It's a little confusing the documentation only mentions two models. I'm new to this codebase (and printer) and it's comforting to see in print what is known to work.

**Comment by KeithCu** [comment]
I decided to buy the Ender 3 Max Neo and sell my Ev2. A custom build, saving the manual configuration, would be awesome.

**Reply by mriscoc** [reply]
You can write a configuration file: https://github.com/mriscoc/Ender3V2S1/wiki/Configuration-files

**Comment by KeithCu** [comment]
The machine arrived. The build is trivial compared to the E3v2,  I did the basic mechanical checking everything I know of. It was well built except the eccentric nuts were too tight.

I installed the professional firmware, with the below configuration file.
However, the bed tramming wizard didn't grab the heights from reasonable positions.
Also, when I would move the X axis to 300 mm, it wouldn't move the full width.
I thought maybe the steps per mm was different, but I saw in the Creality firmware it uses 80 for X and 80.1 for Y, so perhaps it is something else? I checked and the board is a 4.2.2
```gcode
; Professional Firmware Configuration File
;=====================================================
C10                    ; Mark as a configuration file
M117 Applying configuration
;-----------------------------------------------------
M256 B15               ; Set Brightness 
M413 S0                ; Disable Powerloss recovery
M603 L200.00 U200.00   ; Configure Filament Change
C851 S800 M0           ; Probe feedrate and disable multiple probing
C29 L25.00 R275.00 F25.00 B275.00 N5 T50 ; Set Mesh insets, size of 5x5 and bed at 50°C
C100 X0 Y0             ; Set XY Min position to 0
C101 X300 Y300 Z320    ; Set XYZ Max position
C102 X300 Y300         ; Set a bed size of 300x300
C412 M0                ; Filament sensor active mode: LOW
M500                   ; Save all settings
;-----------------------------------------------------
G4 S1                  ; Wait a second
M300 P200              ; Beep
M117 Configuration Applied
```

**Reply by MarkforU** [reply]
Did you edit you machine physical settings in the Advanced menu telling it your xyz dimensions?

**Reply by KeithCu** [reply]
I checked that they were 300mm after applying the config file.

**Reply by mriscoc** [reply]
Set the mesh insets after change machine physical limits

**Comment by DarrenCar** [comment]
How does this look?
```gcode
; Professional Firmware Configuration File
;=====================================================
C10                    ; Mark as a configuration file
M117 Applying configuration
;-----------------------------------------------------
C100 X0 Y0             ; Set XY Min position to 0
C101 X300 Y300 Z320    ; Set XYZ Max position
C102 X300 Y300        ; Set a bed size of 300x300
C29 L25.00 R275.00 F25.00 B275.00 N5 T50 ; Set Mesh insets, size of 5x5 and bed at 50°C
C851 S800 M0           ; Probe feedrate and disable multiple probing
C412 M0                ; Filament sensor active mode: LOW
M256 B150              ; Set Brightness 
M413 S0                ; Disable Powerloss recovery
M603 L200.00 U200.00   ; Configure Filament Change
M500                   ; Save all settings
;-----------------------------------------------------
G4 S1                  ; Wait a second
M300 P200              ; Beep
M117 Configuration Applied
```

**Reply by mriscoc** [reply]
It seems to be good.

**Reply by KeithCu** [reply]
I found on the Max Neo that the X min position is -20. 

I am using this which generally works but might not be correct regarding the Max X. I was having problems with using 280.

```gcode
; Professional Firmware Configuration File
;=====================================================
C10                    ; Mark as a configuration file
M117 Applying configuration
;-----------------------------------------------------
M256 B15               ; Set Brightness 
M413 S0                ; Disable Powerloss recovery
M603 L200.00 U200.00   ; Configure Filament Change
C851 S800 M0           ; Probe feedrate and disable multiple probing
C100 X-20 Y0           ; Set XY Min position to -20, 0
C101 X300 Y300 Z320    ; Set XYZ Max position
C102 X300 Y300         ; Set a bed size of 300x300
C29 L25.00 R275.00 F25.00 B275.00 N5 T50 ; Set Mesh insets, size of 5x5 and bed at 50°C
C412 M0                ; Filament sensor active mode: LOW
M500                   ; Save all settings
;-----------------------------------------------------
G4 S1                  ; Wait a second
M300 P200              ; Beep
M117 Configuration Applied
```

**Reply by RealDogDad** [reply]
Why did you disable powerloss recovery, out of curiousity?

**Reply by cprgolds** [reply]
@RealDogDad - Take a look at this video and you will see a reason to disable power loss recovery.

[https://geekdetour.com/3d-printing/blobs-in-your-3d-prints-power-loss-recovery-can-be-the-problem/](url)

**Comment by ONEEYEWILLIE420** [comment]
thank you so much for this took me a bit to rap my head around the exact things to do but pronter face and your g code worked so far.  bed mesh is off on the low side of x axis and y could go back about 10-15mm back. thanks again to everyone.

**Reply by mriscoc** [reply]
For the custom C-codes you need a special version of Pronterface: https://github.com/mriscoc/Pronterface-PFEd/releases Normal Pronterface will filter any non-standard Marlin G-codes, so if you try to use configuration files with C-codes, they will be removed and only the standard codes will be sent to the printer.

**Reply by ONEEYEWILLIE420** [reply]
the ver. i had was able to accept it. im just learning so takes me days to do what some can do in seconds hahahaha thanks sir.

**Reply by jkaunert** [reply]
version for macOS?

**Comment by Bukkza** [comment]
Hey guys, happy new year everyone! Please help me. I ran into a problem after installing the firmware. The default filament flow is 100%, but when printing, the plastic barely comes out. I have to manually set it to 350%, and then it prints more or less smoothly. My printer is an Ender3 Max Neo, and I used the configuration files from the mriscoc project.

**Reply by mriscoc** [reply]
You need to calibrate your extruder steps https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides
